rubik/solve.py

```diff
@@ -351,6 +354,10 @@ def state8():
             else:
                 self.move("F")
 
+            count += 1
+            if count >= self.inifinite_loop_max_iterations:
```

Comment:
Note: I'm not sure what the minimum value should be for this, but I ran `solve_random_cubes.py` and saw no solve errors for 10000 randomly shuffled cubes.